R/get_national_data.R

```diff
@@ -0,0 +1,155 @@
+#' Get national-level data for countries globally, sourced from the ECDC or WHO.
+#' 
+#' @description Gets raw data using the source-specific function. Includes ISO country codes. 
+#' Then adds columns which were missing from the raw data (calculating cumulative counts from new dailies and vice versa),
+#' cleans and sanitises further. Adds rows and columns of NA values so that data is in a standard format. 
+#' 
+#' @param country Character A string specifying the country to get data from. Not case dependent. Defaults to all countries.
+#' Name should be the English name.
+#' @param totals Boolean. If TRUE, returns totalled data per country up to today's date. If FALSE, returns the full dataset stratified by date and country.
+#' @param source Character A string specifying the data source: "WHO", or "ECDC". Not case dependent. Defaults to ECDC.
+#' @return A tibble with data related to cases, deaths, hospitalisations, recoveries and testing.
+#' @importFrom dplyr %>% group_by arrange select ungroup do everything
+#' @importFrom tidyr drop_na fill
+#' @importFrom tibble tibble
+#' @export
+#' @examples
+#' 
+#' \dontrun{
+#'  get_national_data(country = "canada", totals = TRUE, source = "WHO")
+#' }
+#' 
+
+get_national_data <- function(country = NULL, totals = FALSE, source = "ecdc"){
+  
+  # Error handling -------------------------------------------------------------------
+  if (!is.null(country) & !(is.character(country))){
```

Comment:
I have recently started to use @sbfnk lead and have missing arguments then use missing(country) rather than is NULL. The cost is that it is annoying to pass to internal functions but the pro is that it is a bit cleaner. If passing to another function can do something like if (missing(arg)){arg <- NULL}

Reply:
Great, makes sense. Changed